de.dashup.application/src/main/webapp/views/marketplace.jsp

```diff
@@ -7,17 +9,166 @@
+                                    <p>
+                                        ${fn:escapeXml(mapEntry.key.shortDescription)}
+                                    </p>
+                                </div>
+                                <div class="row" style="margin: 0 auto; width: 164px">
+                                    <div class="col">
+                                        <div class="star-rating" style="font-size: 25px; margin-top: 15px; color: var(--color-dark-gray);">
+                                            <div class="back-stars">
+                                                <i class="fa fa-star" aria-hidden="true"></i>
+                                                <i class="fa fa-star" aria-hidden="true"></i>
+                                                <i class="fa fa-star" aria-hidden="true"></i>
+                                                <i class="fa fa-star" aria-hidden="true"></i>
+                                                <i class="fa fa-star" aria-hidden="true"></i>
+                                                <div class="front-stars"
+                                                     style="width:  ${fn:escapeXml(mapEntry.key.averageRating)}%;color: var(--color-background);">
+                                                    <i class="fa fa-star" aria-hidden="true"></i>
+                                                    <i class="fa fa-star" aria-hidden="true"></i>
+                                                    <i class="fa fa-star" aria-hidden="true"></i>
+                                                    <i class="fa fa-star" aria-hidden="true"></i>
+                                                    <i class="fa fa-star" aria-hidden="true"></i>
+                                                </div>
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                                <div class="row">
+                                    <div class="col" style="text-align: center; width: 100%;">
+                                        <p>"${fn:escapeXml(mapEntry.value.text)}"</p>
+                                        <p>By ${fn:escapeXml(mapEntry.value.userName)} ${fn:escapeXml(mapEntry.value.userSurname)}</p>
+                                    </div>
+                                </div>
+                            </div>
+                        </a>
+                    </c:forEach>
+                </div>
+            </div>
+        </div>
+        <div class="row">
+            <div class="col m6 offset-m3 s10 offset-s1">
+                <h3><fmt:message key="i18n.mostPopular"/></h3>
+                <hr/>
+            </div>
+        </div>
+        <div class="row">
+            <div class="col s12 m6 offset-m3">
+                <div class="row">
+                    <c:forEach items="${mostDownloaded}" var="panel">
```

Comment:
once again you could use a for loop over mostDownloaded and bestRated

Reply:
See line 103: there is the forEach loop??????????????
If u know a better way plz let me know....

Reply:
Yes, you made a JSLT for loop to generate the widgets into the sections mostDownloaded and bestRated. But as these sections are created the same way, you could have created a for loop over the creation of these sections as well. Besides JSLT you can use so called scriplets in JSP as well. They allow you to write normal Java Code in JSP Views, which is one reason, why they are called Java Server Pages. I pushed some sample coding commented at the concerned line, can you implement this real quick?